Lib/fontmake/font_project.py

```diff
@@ -420,6 +450,33 @@ def _output_path(self, ufo, ext, is_instance=False, interpolatable=False,
             os.makedirs(out_dir)
         return os.path.join(out_dir, '%s.%s' % (self._font_name(ufo), ext))
 
+    def _designspace_locations(self, designspace_path):
+        """Map font filenames to their locations in a designspace."""
+
+        maps = []
+        for location_list in varLib.designspace.load(designspace_path):
+            location_map = {}
+            for loc in location_list:
+                abspath = os.path.normpath(os.path.join(
+                    os.path.dirname(designspace_path), loc['filename']))
```

Comment:
can normpath and dirname be hoisted out of this inner loop?  Or are we normalizing the loc['filename'] result as well?